src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper;
+
+    public OrderController(OrderService orderService,
+                           UserService userService,
+                           ShoppingCartService shoppingCartService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+    }
+
+    @GetMapping("/{userId}")
+    public List<OrderResponseDto> getOrdersHistory(@PathVariable Long userId) {
+        List<Order> ordersHistory = orderService.getOrdersHistory(userService.get(userId));
+        return ordersHistory.stream()
+                .map(orderDtoResponseMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/complete/{userId}")
+    public OrderResponseDto completeOrder(@PathVariable Long userId) {
```

Comment:
[The same](https://mate-academy.github.io/jv-program-common-mistakes/java-spring/rest/jv-spring-rest_checklist#common-mistakes-jv-spring-rest)